ACPs/77-reinventing-subnets/README.md

```diff
@@ -46,77 +46,154 @@ For standalone networks like the Avalanche Primary Network, this is done by conn
+Once this transaction is accepted, `AddSubnetValidatorTx` and `RemoveSubnetValidatorTx` are disabled on the Subnet. `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx` must be used to manage the Subnet's validator set going forward.
 
-Subnets are responsible for defining the procedure on how to retrieve the above information from prospective validators.
+### Recovering a Subnet
 
-An EVM Subnet may choose to implement this step like so:
+In the event that a Subnet has no validators, a valid BLS multi-signature cannot be produced. This situation can arise immediately after a Subnet is created or if all the Subnet's validators were removed from the validator set via `SetSubnetValidatorWeightTx` with `Weight = 0`.
 
-- Use the number of tokens the user has staked into a smart contract on the Subnet to determine the weight of their validator
-- Require the user to submit an on-chain transaction with their validator information
-- Generate the Warp message
+A `RecoverSubnetTx` can be used in this situation to instantiate the Subnet's validator set using the `Owner` key defined in `CreateSubnetTx`. In all other situations, the `Owner` key is powerless after a `ConvertSubnetTx` is issued.
+
+Each Subnet Validator will be instantiated with a Balance of `(sum($AVAX inputs) - sum($AVAX outputs) - TxFee) / len(Validators)` rounded down. For a `RecoverSubnetTx` to be valid, `(sum($AVAX inputs) - sum($AVAX outputs) - TxFee) / len(Validators)` must be >= the greater of 5 $AVAX or two weeks of the current fee. This prevents Subnet Validators from being added with too low of an initial balance where they become immediately delinquent based on the continous fee mechanism defined below. A Subnet Validator can leave at any time before the initial $AVAX is consumed and claim the remaining balance to the `ChangeOwner` defined in the transaction.
 
-#### Step 2: Issue a `RegisterSubnetValidatorTx` on the P-Chain
+#### RecoverSubnetTx
+
+```golang
+type SubnetValidator struct {
+    // Must be Ed25519 NodeID
+    NodeID ids.NodeID `json:"nodeID"`
+    // Weight of this validator used when sampling
+    Weight uint64 `json:"weight"`
+    // [Signer] is the BLS key for this validator.
+    // Note: We do not enforce that the BLS key is unique across all validators.
+    //       This means that validators can share a key if they so choose.
+    //       However, a NodeID + Subnet does uniquely map to a BLS key
+    Signer signer.Signer `json:"signer"`
+    // Leftover $AVAX from the [Balance] will be issued to this
+    // owner once it is removed from the validator set.
+    ChangeOwner fx.Owner `json:"changeOwner"`
+}
+
+type RecoverSubnetTx struct {
```

Comment:
maybe Recover is not the best name for the case of first using it, after a convert.  maybe the starting set of validators can be just set on Convert itself?

Reply:
Existing PoA Subnets should be able to keep their validator set through the ConvertSubnetTx so I didn't include the starting set on ConvertSubnetTx.

Recover signifies that this is a transaction that is only used as last-resort. Open to other suggestions

Reply:
It seems the case that is going to be used for the typical flow, not as last resort. Maybe SetInitialSOV?